### PR TITLE
Updating Funcotator tool documentation for 4.1 release.

### LINKED ISSUE #5611: Final Funcotator Documentation Updates for 4.1
Update docs for Funcotator.

Tool documentation and GencodeFuncotationFactory / GencodeFuncotation documents (to create a proper spec. for the gencode annotations).

### PR DESCRIPTION
**Pull request by jonn-smith**
- Added in new funcotator wdl.
- Added more documentation to Funcotator and FuncotatorDataSourceDownloader
- Added Gencode spec to the top-level funcotator tool doc.

Fixes #5611

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java [hunk 1]
```diff
@@ -153,10 +226,466 @@
...
+ *             If the variant does not occur in the expressed transcript of the corresponding gene (e.g. is of type <i>INTRON</i> or <i>IGR</i>), then this field is empty.
+ *         </li>
+ *         <li>
+ *             <b><i>transcriptPos</i></b><br />
+ *             <i>Type: </i>{@link Integer} or Empty<br />
+ *             Position in the chosen transcript of the variant allele.<br />
+ *             For variant alleles that occur at a single base, the format is simply the position at which that variant occurs in the transcript (1-based, inclusive) (e.g. <i>1294</i>)<br />
+ *             For variant alleles spanning multiple bases, the format is:<br />
+ *             <pre>
+ *             [START]_[END]
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: Might want to mention that start and end are inclusive

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java [hunk 2]
```diff
@@ -153,10 +226,466 @@
...
+ *             If the variant does not occur in the expressed transcript of the corresponding gene (e.g. is of type <i>INTRON</i> or <i>IGR</i>), then this field is empty.
+ *         </li>
+ *         <li>
+ *             <b><i>cDnaChange</i></b><br />
+ *             <i>Type: </i>{@link String}<br />
+ *             A {@link String} that summarizes the change resulting from this variant allele in the coding sequence for the transcript in which it occurs.<br />
+ *             Positions in this field are <i>relative to the start of the transcript (1-based, inclusive)</i> unless otherwise noted.<br />
+ *             Generally the format of this field is:<br />
+ *             <pre>
+ *             c.[POSITION][BASES CHANGED]
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: Isn't there a dot or underscore between position and bases changed?

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java [hunk 3]
```diff
@@ -170,8 +699,8 @@
  *
  * <h3>Notes</h3>
  * <ul>
- *     <li>This is a beta tool, and as such may generate errors or warnings.</li>
- *     <li>This tool is the successor to <a href="http://portals.broadinstitute.org/oncotator/">Oncotator</a>, with better support for germline data.</li>
+ *     <li>This tool is the spiritual successor to <a href="http://portals.broadinstitute.org/oncotator/">Oncotator</a>, with better support for germline data, numerous fixes for correctness, and many other features.</li>
+ *     <li>REMEMBER: <strong>Funcotator is NOT Oncotator.</strong></li>
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: Can we get rid of this REMEMBER line?  I do not really think that needs clarification.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java (hunk 1 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: I'll made it more specific.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java (hunk 3 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: I think it's somewhat important to temper expectations for a user that sees Funcotator and assumes it does the exact same thing that Oncotator does.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java (hunk 2 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: That's a general description of the field.  The specifics are in the section following that which breaks down the [POSITION] and [BASES CHANGED] placeholders into the actual format.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java (hunk 3 above)
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java]: I would not worry.  "spiritual successor" encompasses that.  Or verbiage more like "the feature set in Oncotator is not the same as Funcotator.", which is more specific.

**Pull request merged**